src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -821,10 +821,14 @@ public ClusterState execute(ClusterState currentState) {
                             MetaData.Builder metaDataBuilder = MetaData.builder(updatedState.metaData()).removeAllIndices();
                             for (IndexMetaData indexMetaData : updatedState.metaData()) {
                                 IndexMetaData currentIndexMetaData = currentState.metaData().index(indexMetaData.index());
-                                if (currentIndexMetaData == null || currentIndexMetaData.version() != indexMetaData.version()) {
+                                if (currentIndexMetaData == null) {
```

Comment:
I would prefer this instead:

```java
if (currentIndexMetaData != null && currentIndexMetaData.isSameUUID(indexMetaData.uuid()) && currentIndexMetaData.version() == indexMetaData.version())  {
   metaDataBuilder.put(currentIndexMetaData, false);
else {
   metaDataBuilder.put(indexMetaData, false);
}
```

Less code blocks at the price if a longer condition, but this is just my personal style.